src/k8s/cmd/k8s/k8s_bootstrap.go

```diff
@@ -143,21 +158,41 @@ func getConfigFromYaml(filePath string) (apiv1.BootstrapConfig, error) {
 
 func getConfigInteractively(stdin io.Reader, stdout io.Writer, stderr io.Writer) apiv1.BootstrapConfig {
 	config := apiv1.BootstrapConfig{}
-	config.SetDefaults()
 
 	components := askQuestion(
 		stdin, stdout, stderr,
 		"Which components would you like to enable?",
 		componentList,
-		strings.Join(config.Components, ", "),
+		"network, dns, gateway, metrics-server",
 		nil,
 	)
-	config.Components = strings.Split(components, ",")
+	for _, component := range strings.FieldsFunc(components, func(r rune) bool { return unicode.IsSpace(r) || r == ',' }) {
```

Comment:
Can you add a comment for what this is doing ?

Reply:
looks to be splitting the `components` user input on comma or spaces